src/Sylius/Component/Order/Model/Order.php

```diff
@@ -270,8 +258,6 @@ public function removeItem(OrderItemInterface $item)
             $item->setOrder(null);
```

Comment:
This should be removed as item __must__ be assigned to some order, otherwise it's broken.